medusa/metadata/generic.py

```diff
@@ -722,23 +722,22 @@ def _retrieve_show_image(self, image_type, show_obj, which=None):
         indexer_lang = show_obj.lang
 
         try:
-            # There's gotta be a better way of doing this but we don't wanna
-            # change the language value elsewhere
-            lINDEXER_API_PARMS = indexerApi(show_obj.indexer).api_params.copy()
+            if not self.indexer_api:
```

Comment:
We have this same code in 3 different places:
```
if not self.indexer_api:
                # There's gotta be a better way of doing this but we don't wanna
                # change the language value elsewhere
                l_indexer_api_params = indexerApi(show_obj.indexer).api_params.copy()

                l_indexer_api_params['banners'] = True

                if indexer_lang and not indexer_lang == app.INDEXER_DEFAULT_LANGUAGE:
                    l_indexer_api_params['language'] = indexer_lang

                if show_obj.dvdorder != 0:
                    l_indexer_api_params['dvdorder'] = True

                self.indexer_api = indexerApi(show_obj.indexer).indexer(**l_indexer_api_params)

            indexer_show_obj = self.indexer_api[show_obj.indexerid]
```

This could be optimized.